### DISCUSSION TITLE
Bird classification

### DISCUSSION THREAD

**Discussion by chrisb123** [blakeblackshear/frigate#22361] [Ask A Question] [answered]
I tried the built-in bird classification enrichment and too be honest its not good, it usually does not detect the bird species and if it does its usually incorrect
With the addition of sub label classification in 0.17 and if I provide enough training data should this yield better results?

**Comment by inkeep-ai-assistant** [comment, marked as answer]
Yes, using custom classification with sub label assignment should yield better results than the built-in bird classification for your specific use case.[(1)](https://docs.frigate.video/configuration/custom_classification/object_classification)

The built-in bird classification uses a pre-trained MobileNet INat Bird Classification model[(2)](https://docs.frigate.video/configuration/bird_classification), which is a general-purpose model not trained on your specific camera scenes or bird species.

Custom object classification allows you to train a MobileNetV2 classification model on tracked objects using your own image submissions.[(1)](https://docs.frigate.video/configuration/custom_classification/object_classification) For bird classification specifically, you would configure it to classify `bird` objects and set the classification type to `sub_label`.[(1)](https://docs.frigate.video/configuration/custom_classification/object_classification)

Here's how you would configure it:

```yaml
classification:
  custom:
    bird:
      threshold: 0.8
      object_config:
        objects: [bird]
        classification_type: sub_label
```
[(1)](https://docs.frigate.video/configuration/custom_classification/object_classification)[(3)](https://docs.frigate.video/configuration/reference)

The key advantages of custom classification over built-in bird classification are:

- You define the specific bird species classes relevant to your location and cameras[(1)](https://docs.frigate.video/configuration/custom_classification/object_classification)
- The model learns from your actual camera footage and conditions[(1)](https://docs.frigate.video/configuration/custom_classification/object_classification)
- With sufficient training data (at least 100 verified images per camera recommended), accuracy improves significantly[(1)](https://docs.frigate.video/configuration/custom_classification/object_classification)

For sub labels to be assigned, both conditions must be met: each classification attempt must meet the configured threshold (default 0.8), and after at least 3 classification attempts, 60% must agree on the same class label.[(1)](https://docs.frigate.video/configuration/custom_classification/object_classification)

Keep your classes visually distinct to improve accuracy, and include a `none` class for birds that don't fit any specific category.[(1)](https://docs.frigate.video/configuration/custom_classification/object_classification)

📚 **Sources:**
1. [Object Classification](https://docs.frigate.video/configuration/custom_classification/object_classification)
2. [Bird Classification](https://docs.frigate.video/configuration/bird_classification)
3. [Full Reference Config](https://docs.frigate.video/configuration/reference)

---

Have another question? Just tag [@inkeep](https://inkeep.com).

**Comment by leccelecce** [comment]
I have moved to this and after adding a variety of images (around 30 per bird, 7 different species, different poses / angles / lights) it's now working fairly well. Certainly better than the built-in one.

Between 10-20 images it was capable of major false positives but it seems to have stepped up in quality past 30 per bird.

**Reply by chrisb123** [reply]
Good to know.
I only just started, classifications are still terrible, but better than the built-in model.
The camera is capable of higher resolution images, any way to make it use them for classification?
Frigate+ still thinks my dog is a cat or a bird and tiny bollards used to mount glass panels, they are also birds apparently, still need to train the detection  correctly
